zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+
+In order to ensure the security of the transfers, and as we will explain below, we are redefining input dummy notes [#protocol-dummynotes]_ for custom Assets, as we need to enforce that the :math:`\mathsf{AssetId}` of the output note of that split Action is the output of a PRF.
+
+Finally, in this ZIP we also describe the *burn* mechanism, which is a direct extension of the transfer mechanism. The burn process uses a similar mechanism to what is used in Orchard to unshield ZEC, by using the :math:`\mathsf{valueBalance}` of the Asset in question. Burning Assets is useful for many purposes, including bridging of Wrapped Assets and removing supply of Assets.
+
+Specification
+=============
+
+Most of the protocol is kept the same as the Orchard protocol released with NU5, except for the following.
+
+Asset Identifiers
+-----------------
+
+For every new Asset, there must be a new and unique identifier of the Asset. Every Asset is defined by the *Asset description*, :math:`\mathsf{asset\_desc}`, which is a global byte string (valid across all -future- Zcash protocols). From this Asset description the specific Asset identifier, :math:`\mathsf{AssetId}`, is derived with a `GroupHash` and used within the output notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve).
+
+This :math:`\mathsf{AssetId}` will be the base point of the value commitment for the specific custom Asset. Note that the :math:`\mathsf{AssetId}` of the ZEC Asset will be kept as the original value base point, :math:`\mathcal{V}^\mathsf{Orchard}`.
+
+In future network and protocol upgrades, the same Asset description string can be carried on, with potentially a mapping into a different shielded protocol. In that case, the turnstile should know how to transform the Asset identifier, :math:`\mathsf{AssetId}`, from one shielded protocol to another.
+
+Note Structure & Commitment
+---------------------------
+
+First, we need to adapt the components that define the Assets, i.e.: *notes*. A ZSA note differs from an Orchard note [#protocol-notes]_ by additionally including the identifier of the Asset, :math:`\mathsf{AssetId}`. So a ZSA note is a tuple :math:`(\mathsf{g_d, pk_d, v, \rho, \psi, AssetId})`,
+where 
+
+- :math:`\mathsf{AssetId} : \mathbb{P}` is the unique random element of the Pallas group [#protocol-pallasandvesta]_ that identifies each Asset in the Orchard protocol. 
```

Comment:
```suggestion
- :math:`\mathsf{AssetId} : \mathbb{P*}` is the unique random element of the Pallas group [#protocol-pallasandvesta]_ that identifies each Asset in the Orchard protocol, a valid non-bottom group element that is not the identity.
```

If it did include the identity, it would not be binding

Reply:
I noted during the ZIP editors review that if an asset can be produced with an `AssetId` of the identity, the value balance for that asset becomes trivially satisfiable (independently of any other asset), making counterfeiting of the asset inherent to its identity 🥁🔔

Non-identity `AssetId`s don't have this issue; there we still rely on having unknown relationships to other `AssetId`s, but within a single asset this is sufficient to ensure the value balance is binding.

Reply:
As discussed, the probability for the identity point for `pallas::Point::hash_to_curve()` is negligible. 
We will add a consensus rule for the issuance bundle to ensure that the identity point isn't used as `assetBase`.

Reply:
We discussed this in a recent sync, the conclusion was to explicitly preclude the identity, as given in the suggestion. This has been incorporated.